docs: Revamp API Platform Admin docs (#2141)
api-platform/docs: admin/customizing.md

+       <ResourceGuesser name="books" list={BookList} />
        <ResourceGuesser name="reviews" />
    </HydraAdmin>
);
```

Likewise, you can use the `icon` prop to customize the icon displayed in the menu:

```diff
+import AutoStoriesIcon from '@mui/icons-material/AutoStories';
+import ReviewsIcon from '@mui/icons-material/Reviews';
import { HydraAdmin, ResourceGuesser } from "@api-platform/admin";

const App = () => (
    <HydraAdmin entrypoint={...}>
-       <ResourceGuesser name="books" />
+       <ResourceGuesser name="books" icon={AutoStoriesIcon} />
-       <ResourceGuesser name="reviews" />
+       <ResourceGuesser name="reviews" icon={ReviewsIcon} />
    </HydraAdmin>
);
```

Here is the result:

![Admin menu with custom icons](./images/admin-menu-icons.png)

## Customizing the `<ListGuesser>`

By default, `<ResourceGuesser>` will render a `<ListGuesser>` component as the list view for a resource.

Again, this component will automatically introspect the API schema and generate a list view with all the fields of the resource.

![Admin default generated list view](./images/admin-default-list.png)

This is already usable, but may not provide the best user experience yet.

> Here, too, to start customizing the list view, you can look at the DevTools console. You will see messages like this:

Reviewer: ```suggestion
To start customizing the list view, you can look at the DevTools console. You will see messages like this:
```

However in some cases you may want to go further and use a React Admin [field components](https://marmelab.com/react-admin/Fields.html), such as [`<TextField>`](https://marmelab.com/react-admin/TextField.html), [`<DateField>`](https://marmelab.com/react-admin/DateField.html) or [`<ReferenceField>`](https://marmelab.com/react-admin/ReferenceField.html) directly, to access more advanced features.

For instance, you can replace a `<FieldGuesser>` with a [`<DateField>`](https://marmelab.com/react-admin/DateField.html) to control more precisely how the publication date is displayed, leveraging the [`showTime`](https://marmelab.com/react-admin/DateField.html#showtime) prop:

```diff
import { ShowGuesser, FieldGuesser } from '@api-platform/admin';
+import { DateField } from 'react-admin';

const ReviewShow = () => (
  <ShowGuesser>
-     <FieldGuesser source="publicationDate" />
+     <DateField showTime source="publicationDate" />
  </ShowGuesser>
);
```

## Customizing the `<EditGuesser>` and `<CreateGuesser>`

Customizing the `<EditGuesser>` and `<CreateGuesser>` is very similar to customizing the `<ShowGuesser>`.

> Again, we can start by looking at the DevTools console to get the initial code of the components.

Reviewer: ```suggestion
We can start by looking at the DevTools console to get the initial code of the components.
```

        <InputGuesser source="book" />
    </EditGuesser>
);

Then, update your main admin component:

import { HydraAdmin, ResourceGuesser } from "@api-platform/admin";
import { ReviewEdit } from './ReviewEdit';

const App = () => (
    <HydraAdmin entrypoint={...}>
        <ResourceGuesser name="reviews" edit={ReviewEdit} />
        {/* ... */}
    </HydraAdmin>
);
```

Let's customize this `ReviewEdit` component to:

- reorder the inputs
- make the `body` input multiline
- mark the `publicationDate` input as read-only

```diff
export const ReviewEdit = () => (
    <EditGuesser>
-       <InputGuesser source="rating" />
-       <InputGuesser source="body" />
-       <InputGuesser source="author" />
-       <InputGuesser source="publicationDate" />
-       <InputGuesser source="book" />
+       <InputGuesser source="author" />
+       <InputGuesser source="book" />
+       <InputGuesser source="body" multiline />
+       <InputGuesser source="rating" />
+       <InputGuesser source="publicationDate" readOnly />
    </EditGuesser>
);
```

Here is the result:

![Admin with customized edit guesser](./images/admin-custom-edit-guesser.png)

> **Tip:** Here, we leveraged the `multiline` and `readOnly` props of the `<InputGuesser>` component. But you can use any [common input prop](https://marmelab.com/react-admin/Inputs.html#common-input-props) supported by react-admin [Inputs](https://marmelab.com/react-admin/Inputs.html) on them.

Reviewer: ```suggestion
**Tip:** Here, we leveraged the `multiline` and `readOnly` props of the `<InputGuesser>` component. But you can use any [common input prop](https://marmelab.com/react-admin/Inputs.html#common-input-props) supported by React Admin [Inputs](https://marmelab.com/react-admin/Inputs.html) on them.
```

> **Tip:** Here, we leveraged the `multiline` and `readOnly` props of the `<InputGuesser>` component. But you can use any [common input prop](https://marmelab.com/react-admin/Inputs.html#common-input-props) supported by react-admin [Inputs](https://marmelab.com/react-admin/Inputs.html) on them.

Reviewer: ```suggestion
**Tip:** Here, we leveraged the `multiline` and `readOnly` props of the `<InputGuesser>` component. But you can use any [common input prop](https://marmelab.com/react-admin/Inputs.html#common-input-props) supported by React Admin [Inputs](https://marmelab.com/react-admin/Inputs.html) on them.
```

This leverages both [`<ReferenceInput>`](https://marmelab.com/react-admin/ReferenceInput.html) and [`<AutocompleteInput>`](https://marmelab.com/react-admin/AutocompleteInput.html) to offer a better user experience when editing the relation:

```diff
import { EditGuesser, InputGuesser } from '@api-platform/admin';
+import { ReferenceInput, AutocompleteInput } from 'react-admin';

const ReviewsEdit = () => (
  <EditGuesser>
    <InputGuesser source="author" />
-   <InputGuesser source="book" />
+   <ReferenceInput source="book" reference="books">
+     <AutocompleteInput
+       filterToQuery={(searchText) => ({ title: searchText })}
+       optionText="title"
+     />
+   </ReferenceInput>
  </EditGuesser>
);
```

![Admin With AutocompleteInput](./images/AutocompleteInput.png)

> [!WARNING]
> When replacing `<InputGuesser>` with a React Admin input component, the validation rules are not automatically applied. You will need to manually add them back. Fortunately, this is very easy to do. Read the [Validation With React Admin Inputs](./validation.md#validation-with-react-admin-inputs) section to learn more.

## Next Step

> The above examples limit to customizing the various API Platform Admin Guessers, but this is just the tip of the iceberg.

Reviewer: ```suggestion
The above examples are limited to customizing the various API Platform Admin Guessers, but this is just the tip of the iceberg.
```
